iam_role.go

```diff
@@ -249,6 +249,65 @@ func (c *Client) CreateIamTrustRole(options *CreateIamRoleOptions) (*IamRoleResp
 	return cr, nil
 }
 
+func jsonDecode(request *IamRoleRequest, c *Client, operation string) (obj []byte, err error) {
+	obj, e := json.Marshal(struct {
+		IamRoleRequest
+		AccountDetails
+	}{*request, c.AccountDetails})
+
+	if e != nil {
+		err = fmt.Errorf("Error encoding IAM %s JSON: %s", operation, e)
+	}
+	return
+}
+
+func updateOptions(o *IamRoleOptions) (options *IamRoleRequest, err error) {
+	if o.RoleName == nil {
+		err = fmt.Errorf("RoleName option must not be nil")
+	}
+	if o.Tags == nil {
+		err = fmt.Errorf("Tags option must not be nil")
+	}
+	options = &IamRoleRequest{
+		RoleName: *o.RoleName,
+		Tags:     *o.Tags,
+	}
+	return
+}
+
+// UpdateIamRole adds resource tags to an existing IAM role.
+func (c *Client) UpdateIamRole(options *IamRoleOptions) (*IamRoleResponse, error) {
+	requestOptions, _ := updateOptions(options)
+	log.Printf("[INFO] Updating IAM role %s with Tags: %v", requestOptions.RoleName, requestOptions.Tags)
+
+	const operation string = "Update Roles"
+
+	b, _ := jsonDecode(requestOptions, c, operation)
```

Comment:
If you are going to silently ignore errors returned by this method then you should strongly consider _not_ returning an error type at all.  I recommend properly handling the error here though, failing to serialize your request body (though unlikely) would result in strange behavior against the API at best, and at worst would create hard to diagnose errors for the consumer.